components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java

```diff
@@ -1599,6 +1600,30 @@ public static boolean jsFunction_addAPISubscription(Context cx,
             trustedIdp = args[7].toString();
         }
 
+        APIConsumer apiConsumer = getAPIConsumer(thisObj);
```

Comment:
Keep line 1603-1625 in a separate private method (#addSubscriber) and call it here.